src/risk_distributions/risk_distributions.py

```diff
@@ -45,79 +46,66 @@ def _get_optimization_result(data: pd.DataFrame, func: Callable,
-    distributions.
-
-    Parameters
-    ---------
-    data :
-        Table where each row has a `mean` and `standard_deviation` for a single distribution.
-    distribution :
-        The distribution class to use to calculate parameters.
-
-    Returns
-    --------
-        A dictionary containing the parameters, as well as the min and max range data.
-    """
-    ranges = distribution._get_min_max(data)
-    params = distribution._get_params(data, ranges)
-
-    return {**ranges, **params}
+    if mean is not None and len(mean) != len(sd):
+        raise ValueError(f"You must specify mean and standard deviation for the same number of distributions. You "
+                         f"specified {len(mean)} mean values and {len(sd)} standard deviation values.")
 
 
 class BaseDistribution:
     """Generic vectorized wrapper around scipy distributions."""
 
     distribution = None
 
-    def __init__(self, data: pd.DataFrame=None, params: Dict[str, Union[np.ndarray, pd.Series]]=None,
-                 mean: Union[pd.Series, float, int]=None, std_dev: Union[pd.Series, float, int]=None):
+    def __init__(self, params: pd.DataFrame=None, mean: Union[pd.Series, float, int]=None,
+                 sd: Union[pd.Series, float, int]=None):
 
-        validate_parameters(data, params, mean, std_dev)
+        validate_parameters(params, mean, sd)
 
-        if (mean and std_dev) or data is not None:
-            if mean:
-                data = pd.DataFrame({'mean': mean, 'standard_deviation': std_dev})
-            self._range = self._get_min_max(data)
-            self._parameter_data = self._get_params(data, self._range)
+        if mean is not None:
+            self._parameter_data = self.get_params(mean, sd)
         else:
-            self._range = {k: v for k, v in params.items() if k in ('x_min', 'x_max')}
-            self._parameter_data = {k: v for k, v in params.items() if k not in ('x_min', 'x_max')}
-
+            self._parameter_data = params
 
     @staticmethod
-    def _get_min_max(data: pd.DataFrame) -> Dict[str, pd.Series]:
+    def _get_min_max(mean: pd.Series, sd: pd.Series) -> pd.DataFrame:
         """Gets the upper and lower bounds of the distribution support."""
-        data_mean, data_sd = data['mean'], data['standard_deviation']
-        alpha = 1 + data_sd ** 2 / data_mean ** 2
-        scale = data_mean / np.sqrt(alpha)
+        alpha = 1 + sd ** 2 / mean ** 2
+        scale = mean / np.sqrt(alpha)
         s = np.sqrt(np.log(alpha))
         x_min = stats.lognorm(s=s, scale=scale).ppf(.001)
         x_max = stats.lognorm(s=s, scale=scale).ppf(.999)
-        return {'x_min': pd.Series(x_min, index=data.index),
-                'x_max': pd.Series(x_max, index=data.index)}
+        return pd.DataFrame({'x_min': x_min, 'x_max': x_max}, index=mean.index)
 
+    @classmethod
+    def get_params(cls, mean: Union[pd.Series, float, int], sd: Union[pd.Series, float, int]) -> pd.DataFrame:
```

Comment:
the reason it's a class method is because we use it in vph to calculate the params ahead of time so we can interpolate over them. That's also why there's an option to pass pre-calculated params to the constructor. I have it as a class method rather than a static method because I couldn't see a way to call the private _get_params of each child class without having a cls object I can use to say `cls._get_params`

Reply:
Oh wait are you asking why I have a public get_params on the parent class instead of just having one on each child? That was because I wanted to be able to have the private methods assume that they were always getting a series rather than having to check if it were an individual number. But maybe that's not necessary. I can push that down the individual _get_params where's it needed

Reply:
Just updated it to only have a public get_params that is only implemented in subclasses

Reply:
yeah the second response was what I was thinking of. Thanks for explaining, I was just wondering for my own edification